index.js

```diff
@@ -239,32 +271,43 @@ class PearIPC extends ReadyResource {
   }
 
   unref () {
-    if (this.rawStream?.unref) this.rawStream.unref()
+    this._heartbeat?.unref()
+    this._timeout?.unref()
+    this.rawStream?.unref()
     this.server?.unref()
   }
 
   ref () {
-    if (this.rawStream?.ref) this.rawStream.ref()
+    this._heartbeat?.ref()
+    this._timeout?.ref()
+    this.rawStream?.ref()
```

Comment:
Sanity check: this has different behaviour than before. Is `ref` guaranteed to exist on a `rawStream`? (previous version didn't assume that)